kasa/cli.py

```diff
@@ -543,6 +543,23 @@ def _schedule_list(dev, type):
         click.echo(f"No rules of type {type}")
 
 
+@schedule.command(name="delete_rule")
+@pass_dev
+@click.option("--id", type=str, required=True)
+async def delete_rule(dev, id):
+    """Delete rule from device."""
+    schedule = dev.modules["schedule"]
+    rule_to_delete = ""
+    for rule in schedule.rules:
+        if rule.id == id:
+            rule_to_delete = rule
```

Comment:
```suggestion
    rule_to_delete = next(filter(lamda rule: rule.id == id), None)
```
just a suggestion about potential alternative, but it's also okay as it is (but initialize to `None` instead of `""`).

Reply:
thanks for the feedback @rytilahti! The suggestion didn't include the rules list so I updated the line and pushed.